eip-5564.md

```diff
@@ -8,174 +8,170 @@ status: Draft
+- This function performs the following computations:
+  - A shared secret $s$ is computed as $s = p_{ephemeral} \cdot P_{view}$.
+  - The entropy is encrypted with $e_{encrypted} = s \oplus r$.
+  - The sender's ephemeral public key is computed as $P_{ephemeral} = p_{ephemeral} \cdot G$.
+  - The recipient's stealth address is computed as $a_{stealth} = P_{spend} \cdot r$.
+  - The method returns $a_{stealth}$, $s$, $keccak256(s)$, $e_{encrypted}$, and $P_{ephemeral}$.
+
+Sending funds now works as follows:
+- Sender uses the contract of their choice to send something to $a_{stealth}$, and provides $e_{encrypted}$, $P_{ephemeral}$, and any other metadata to the send method.
+- The contract calls `IERC5564Messenger.announce` with $a_{stealth}$, $e_{encrypted}$, $P_{ephemeral}$, and any metadata.
+
+To scan for funds, a recipient must retrieve all logs from the `IERC5564Messenger` contract.
+They then check if they can compute the private key $p_{stealth}$ that derives to the emitted stealth address $a_{stealth}$, using the other emitted parameters.
+(This is pretty much the same as how Umbra works now, but perhaps we add [view tags](https://github.com/ScopeLift/umbra-protocol/issues/377) to the announcement event to make this faster).
 
 ## Rationale
 This EIP emerged from the need of having privacy-preserving ways to transfer ownership without revealing the recipient's identity. Tokens can reveal sensitive private information about the owner. While users might want to prove the ownership of an NFT concert ticket, they might not want to reveal personal account-related information at the same time. The standardization of stealth address generation represents a significant effort for privacy. Privacy-preserving solutions require standards to gain adoption, therefore it is critical to focus on generalizable ways of implementing related solutions.
 
 This extension standardizes the method to generate and look up stealth addresses. Users can send assets without having to interact with the recipient beforehand. Furthermore, users can verify if they have been the recipient of a transfer without requiring interactions with the chain. Stealth addresses allow only the recipients of token transfers to see that they were the recipients.
 
-The authors identify the trade-off between on- and off-chain efficiency: Althoug, including a Monero-like `view tags` mechanism might help recipients to parse announcements more quickly, it adds complexity to the announcement event.
-
-The address stealthAddressRecipient MUST be included in the announcement event, allowing users to quickler verify ownership without having to query the chain for positive account balances.
+The authors identify the trade-off between on- and off-chain efficiency: Although, including a Monero-like `view tags` mechanism might help recipients to parse announcements more quickly, it adds complexity to the announcement event.
```

Comment:
This is another decision we still have to make. Maybe a good start is running some benchmarks to compare computation time? We can take all logs emitted from Umbra on polygon (where there's the most usage), append view tags, then benchmark the time to scan through all with and without view tags